File: csharp-client/src/main/csharp/Rakam/Client/Model/Recipe.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Rakam.Client.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class Recipe {
    
    /// <summary>
    /// Gets or Sets Strategy
    /// </summary>
    [DataMember(Name="strategy", EmitDefaultValue=false)]
    public string Strategy { get; set; }

    
    /// <summary>
    /// Gets or Sets Project
    /// </summary>
    [DataMember(Name="project", EmitDefaultValue=false)]
    public string Project { get; set; }

    
    /// <summary>
    /// Gets or Sets Collections
    /// </summary>
    [DataMember(Name="collections", EmitDefaultValue=false)]
    public Dictionary<string, Collection> Collections { get; set; }

    
    /// <summary>
    /// Gets or Sets Reports
    /// </summary>
    [DataMember(Name="reports", EmitDefaultValue=false)]
    public List<ReportBuilder> Reports { get; set; }

    
    /// <summary>
    /// Gets or Sets MaterializedViews
    /// </summary>
    [DataMember(Name="materializedViews", EmitDefaultValue=false)]
    public List<MaterializedViewBuilder> MaterializedViews { get; set; }

    
    /// <summary>
    /// Gets or Sets ContinuousQueries
    /// </summary>
    [DataMember(Name="continuousQueries", EmitDefaultValue=false)]
    public List<ContinuousQueryBuilder> ContinuousQueries { get; set; }

    
    /// <summary>
    /// Gets or Sets ContinuousQueryBuilders
    /// </summary>
    [DataMember(Name="continuousQueryBuilders", EmitDefaultValue=false)]
    public List<ContinuousQueryBuilder> ContinuousQueryBuilders { get; set; }

    
    /// <summary>
    /// Gets or Sets MaterializedViewBuilders
    /// </summary>
    [DataMember(Name="materializedViewBuilders", EmitDefaultValue=false)]
    public List<MaterializedViewBuilder> MaterializedViewBuilders { get; set; }

    

    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Recipe {\n");
      
      sb.Append("  Strategy: ").Append(Strategy).Append("\n");
      
      sb.Append("  Project: ").Append(Project).Append("\n");
      
      sb.Append("  Collections: ").Append(Collections).Append("\n");
      
      sb.Append("  Reports: ").Append(Reports).Append("\n");
      
      sb.Append("  MaterializedViews: ").Append(MaterializedViews).Append("\n");
      
      sb.Append("  ContinuousQueries: ").Append(ContinuousQueries).Append("\n");
      
      sb.Append("  ContinuousQueryBuilders: ").Append(ContinuousQueryBuilders).Append("\n");
      
      sb.Append("  MaterializedViewBuilders: ").Append(MaterializedViewBuilders).Append("\n");
      
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: java-client/src/main/java/org/rakam/client/model/Recipe.java
package org.rakam.client.model;

import org.rakam.StringUtil;
import org.rakam.client.model.MaterializedViewBuilder;
import org.rakam.client.model.ContinuousQueryBuilder;
import org.rakam.client.model.ReportBuilder;
import org.rakam.client.model.Collection;
import java.util.*;
import java.util.Map;



import io.swagger.annotations.*;
import com.fasterxml.jackson.annotation.JsonProperty;


@ApiModel(description = "")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2015-11-19T22:59:17.166+02:00")
public class Recipe   {
  

public enum StrategyEnum {
  DEFAULT("DEFAULT"),
  SPECIFIC("SPECIFIC");

  private String value;

  StrategyEnum(String value) {
    this.value = value;
  }

  @Override
  public String toString() {
    return value;
  }
}

  private StrategyEnum strategy = null;
  private String project = null;
  private Map<String, Collection> collections = new HashMap<String, Collection>();
  private List<ReportBuilder> reports = new ArrayList<ReportBuilder>();
  private List<MaterializedViewBuilder> materializedViews = new ArrayList<MaterializedViewBuilder>();
  private List<ContinuousQueryBuilder> continuousQueries = new ArrayList<ContinuousQueryBuilder>();
  private List<ContinuousQueryBuilder> continuousQueryBuilders = new ArrayList<ContinuousQueryBuilder>();
  private List<MaterializedViewBuilder> materializedViewBuilders = new ArrayList<MaterializedViewBuilder>();

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("strategy")
  public StrategyEnum getStrategy() {
    return strategy;
  }
  public void setStrategy(StrategyEnum strategy) {
    this.strategy = strategy;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("project")
  public String getProject() {
    return project;
  }
  public void setProject(String project) {
    this.project = project;
  }

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("collections")
  public Map<String, Collection> getCollections() {
    return collections;
  }
  public void setCollections(Map<String, Collection> collections) {
    this.collections = collections;
  }

  
  /**
   **/
  @ApiModelProperty(required = true, value = "")
  @JsonProperty("reports")
  public List<ReportBuilder> getReports() {
    return reports;
  }
  public void setReports(List<ReportBuilder> reports) {
    this.reports = reports;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("materializedViews")
  public List<MaterializedViewBuilder> getMaterializedViews() {
    return materializedViews;
  }
  public void setMaterializedViews(List<MaterializedViewBuilder> materializedViews) {
    this.materializedViews = materializedViews;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("continuousQueries")
  public List<ContinuousQueryBuilder> getContinuousQueries() {
    return continuousQueries;
  }
  public void setContinuousQueries(List<ContinuousQueryBuilder> continuousQueries) {
    this.continuousQueries = continuousQueries;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("continuousQueryBuilders")
  public List<ContinuousQueryBuilder> getContinuousQueryBuilders() {
    return continuousQueryBuilders;
  }
  public void setContinuousQueryBuilders(List<ContinuousQueryBuilder> continuousQueryBuilders) {
    this.continuousQueryBuilders = continuousQueryBuilders;
  }

  
  /**
   **/
  @ApiModelProperty(value = "")
  @JsonProperty("materializedViewBuilders")
  public List<MaterializedViewBuilder> getMaterializedViewBuilders() {
    return materializedViewBuilders;
  }
  public void setMaterializedViewBuilders(List<MaterializedViewBuilder> materializedViewBuilders) {
    this.materializedViewBuilders = materializedViewBuilders;
  }

  

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Recipe {\n");
    
    sb.append("    strategy: ").append(StringUtil.toIndentedString(strategy)).append("\n");
    sb.append("    project: ").append(StringUtil.toIndentedString(project)).append("\n");
    sb.append("    collections: ").append(StringUtil.toIndentedString(collections)).append("\n");
    sb.append("    reports: ").append(StringUtil.toIndentedString(reports)).append("\n");
    sb.append("    materializedViews: ").append(StringUtil.toIndentedString(materializedViews)).append("\n");
    sb.append("    continuousQueries: ").append(StringUtil.toIndentedString(continuousQueries)).append("\n");
    sb.append("    continuousQueryBuilders: ").append(StringUtil.toIndentedString(continuousQueryBuilders)).append("\n");
    sb.append("    materializedViewBuilders: ").append(StringUtil.toIndentedString(materializedViewBuilders)).append("\n");
    sb.append("}");
    return sb.toString();
  }
}
